common/web/types/test/ldml-keyboard/test-ldml-keyboard-xml-reader.ts

```diff
@@ -129,3 +142,32 @@ describe('ldml keyboard xml reader tests', function () {
     },
   ]);
 });
+
+describe('check scan code routines', () => {
+  it('should be able to detect bad single scancodes', () => {
+    const badScans = new Set<number>();
+    const ckey = CLDRScanToVkey(0xFF, badScans);
+    assert.isUndefined(ckey, `expected undefined 0xFF`);
+    assert.equal(badScans.size, 1);
+    assert.sameDeepMembers(Array.from(badScans.values()), [0xFF]);
+  });
+  it('should be able to detect bad list scancodes', () => {
+    const badScans = new Set<number>();
+    const ckeys = CLDRScanToKeyMap([[0x02, 0xFF]], badScans);
+    assert.sameDeepMembers(ckeys, [[49, undefined]]);
+    assert.equal(badScans.size, 1);
+    assert.sameDeepMembers(Array.from(badScans.values()), [0xFF]);
+  });
+  it('CLDRScanToUSVirtualKeyCodes should be 1:1', () => {
+    const vkeyToScan = new Map<number, number>();
+    // check all scancodes
+    for (let scan = 0; scan <= 0xFF; scan++) {
+      const vkey = CLDRScanToVkey(scan);
+      if (vkey !== undefined && vkey !== 226) {
```

Comment:
Can we use the defined constant here and add a comment to explain why we ignore K_oE2? And in fact, can we be more explicit about the test -- scan should be either 0x56 or 0x7D:

```suggestion
      if ((scan === 0x56 || scan === 0x7D) && vkey === USVirtualKeyCodes.K_oE2) {
        // We allow dup of K_oE2 / 0x56 iso,abnt2 / 0x7D - jis
        continue;
      }
      if (vkey !== undefined) {
```

Reply:
sorry i missed your suggestion.  in any event, my code catches if some other scan code duplicates K_oE2

Reply:
Your changes look good.